### ISSUE TITLE
How can we use DPM++ 2M Karras sampler in Diffusers?

### ISSUE THREAD

**Issue by danishboy000** [huggingface/diffusers#1887]
I'm trying to use the latest samplers which provide better performance with stable diffusion in the diffusers library, but I couldn't find DPM++ 2M Karras sampler in the library. These samplers are present in Automatic111. Can anyone guide me on how to use these samplers in the Diffusers library?

**Comment by patrickvonplaten** [comment]
> I'm trying to use the latest samplers which provide better performance with stable diffusion in the diffusers library, but I couldn't find DPM++ 2M Karras sampler in the library. These samplers are present in Automatic111. Can anyone guide me on how to use these samplers in the Diffusers library?

Hey, this should be identical to this scheduler: https://huggingface.co/docs/diffusers/api/schedulers/multistep_dpm_solver

Could you try using this one to see if quality matches the one of Auto1111.

**Comment by danishboy000** [comment]
I tried this and the quality didn't match.

**Comment by pcuenca** [comment]
I *think* @danishboy000 might be referring to the use of Karras sigmas #1633 in combination with this scheduler's algorithm, but I'm not an expert in Automatic1111 nomenclature. Perhaps @apolinario can confirm?

**Comment by apolinario** [comment]
Yes that is correct!

**Comment by patrickvonplaten** [comment]
@danishboy000 could you copy-paste some code snippets that you used for Auto1111 and `diffusers` for comparison?

**Comment by apolinario** [comment]
@patrickvonplaten , I think in this issue https://github.com/huggingface/diffusers/issues/1633 I showcase the code snippet used by AUTO1111 for using Karras sigmas for all samplers

**Comment by Stax124** [comment]
@patrickvonplaten I was able to get KDiffusion samplers with Karras sigmas with some modifications to your KDiffusion experimental model:

https://github.com/Stax124/voltaML-fast-stable-diffusion/blob/4b6a23c5a09ee6375b6dbc86ba1d207df80eea7b/core/diffusers/kdiffusion/KDiffusionModel.py#L513

I am getting similar results to A111 (using AnythingV4 - there are multiple versions of the checkpoint, hence the tiny change, quality-wise seems good to me)

Volta:
![download (2)](https://user-images.githubusercontent.com/60222162/213871430-4b910e0f-9063-4d13-ad5d-bfe905b05be1.png)

A111:
![00000-420](https://user-images.githubusercontent.com/60222162/213871482-fec980e5-f0dc-48ad-9c28-ec76d819a922.png)

I would make a PR but I am busy right now, so if you have someone that can take a look at it, feel free to add it.
If not, I will create a PR in the future, but no idea when that will happen.

PS - There are more optimizations in the model, callback was implemented as well.

Thanks for the hard work on `diffusers`, we really appreciate it.
